src/internal/clusterstate/v2.5.0/collection.go

```diff
@@ -267,32 +266,84 @@ func migratePostgreSQLCollection(ctx context.Context, tx *pachsql.Tx, name strin
 		vals[oldKey] = pair{newKey, proto.Clone(newVal)}
 
 	}
-	log.Info(ctx, fmt.Sprintf("Migrating collection.%s", name))
-	i := 0
+
+	// Optimized code for building the UPDATEs and sending them to Postgres in batches
+	// NOTE: This code necessitated upgrading from sqlx 1.2.0 to sqlx 1.3.x to get batching support for named variables
+
+	log.Info(ctx, fmt.Sprintf("Migrating %s [%d rows]", name, len(vals)))
+
+	batch_size := 1000 // Maximum size of each batch
+	batch_count := 0   // Incrementing count of the number of rows in the current batch
+	total_count := 0   // Incrementing count of the total number of rows
+	query := ""        // String to hold the UPDATE query
```

Comment:
I would use a `var ( … )` block for this. Opinions vary, but I think that it would be clearer.